lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -185,6 +242,27 @@ defmodule Trento.Domain.SapSystem do
     |> Multi.execute(&maybe_emit_sap_system_health_changed_event/1)
   end
 
+  # Restore sap system
+  # Same registration rules
+  def execute(
+        %SapSystem{deregistered_at: deregistered_at} = sap_system,
+        %RegisterApplicationInstance{} = instance
+      )
+      when not is_nil(deregistered_at) do
+    sap_system
+    |> Multi.new()
+    |> Multi.execute(fn sap_system ->
+      emit_application_instance_registered_or_application_instance_health_changed(
+        sap_system,
+        instance
+      )
+    end)
+    |> Multi.execute(fn sap_system ->
+      maybe_emit_sap_system_restored_event(sap_system, instance)
+    end)
+    |> Multi.execute(&maybe_emit_sap_system_health_changed_event/1)
```

Comment:
Why do you need this `multi`?
The restoration already changes the health, right?
And once it is restored, it wouldn't come from this flow
If it is for broadcasting messages, I think the restoration one should be enough

Reply:
You're right, thanks for catching this